### ISSUE TITLE
change 6.x default mapping from _default_ to doc

### ISSUE THREAD

**Issue by jsvd** [logstash-plugins/logstash-output-elasticsearch#696]
as the `_default_` mapping will be deprecated in elasticsearch 6.x, the [mapping used for this elasticsearch version](https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/master/lib/logstash/outputs/elasticsearch/elasticsearch-template-es6x.json#L8) should be changed as well.

for more context please check elasticsearch's issue on removing types: https://github.com/elastic/elasticsearch/issues/15613

**Comment by jpountz** [comment]
What would be a reasonable timeframe for this change? It would help move forward with types removal in Elasticsearch 7.0.

**Comment by jpountz** [comment]
@andrewvc Elasticsearch will need to remove support for the `_default_` mapping in master in order to move forward with the removal of types. Could you give me an ETA for this change?

**Comment by jsvd** [comment]
I have created a PR to add the template for 7.x with the _default_ mapping changed to doc https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/713

**Comment by fedelemantuano** [comment]
I've just read this article [schedule_for_removal_of_mapping_types](https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html#_schedule_for_removal_of_mapping_types), as best practice suggests:

```
The preferred type name is _doc, so that index APIs have the same path as they will have in 7.0: PUT {index}/_doc/{id} and POST {index}/_doc
```
For me the default should be `_doc`. In my configuration I had to use `document_type => "_doc"`, but I should not use `document_type` anymore.

**Comment by andrewvc** [comment]
Closing due to this being merged in #713 

@fedelemantuano there is no need to do anything before 7.x
